vendor/guava-20_0/build.gradle

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * License); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+apply plugin: org.apache.beam.gradle.VendorJavaPlugin
+
+description = "Apache Beam :: Vendored Dependencies :: Guava 20"
+
+group = "org.apache.beam"
+version = "0.1"
+
+
+vendorJava(
+  dependency: "com.google.guava:guava:20.0",
+  packagesToRelocate: ["com.google.common", "com.google.thirdparty"],
```

Comment:
Since we are always relocating everything, is there a reason to enumerate these and not just put ""?

It seems like this could be something that we force for all packages to not allow "partial" relocations, the only reason I had enumerated the packages to relocate before was that under the old style we were relocating each package into a different package prefix which we don't need here.

From the shadow docs, `Relocation operates at a package level. It is not necessary to specify any patterns for matching, it will operate simply on the prefix provided.`

Reply:
That is what I want, too. But since relocation is independent from jars, I think `""` will also rewrite all the `java.` imports and classes. In cases more complex than Guava, this applies to all dependencies.

Reply:
That is unfortunate but good to know going forward. At least the test will make sure we don't do something silly.